pkg/api/bzzchunk.go

```diff
@@ -28,11 +35,46 @@ func (s *server) chunkUploadHandler(w http.ResponseWriter, r *http.Request) {
 	address, err := swarm.ParseHexAddress(addr)
 	if err != nil {
 		s.Logger.Debugf("bzz-chunk: parse chunk address %s: %v", addr, err)
-		s.Logger.Error("bzz-chunk: error uploading chunk")
+		s.Logger.Error("bzz-chunk: parse chunk address")
 		jsonhttp.BadRequest(w, "invalid chunk address")
 		return
 	}
 
+	// if tag header is not there create a new one
+	var tag *tags.Tag
+	tagUidStr := r.Header.Get(TagHeaderUid)
+	if tagUidStr == "" {
+		tagName := fmt.Sprintf("unnamed_tag_%d", time.Now().Unix())
```

Comment:
i suggest that if there is no longer a possibility to name tags (this was the original tag header purpose, which is now removed), then the tag name field should be removed from the codebase, since we're ending up with unnamed tags everywhere. either this or reinstate the header with possibility to give a name to a tag (a name which is not unique, as opposed to the uid that is generated there). it was not my intention that this functionality be removed. you are free to iterate on this in later pr if you want to.

Reply:
we now have two ways to use tags
- we have a seperate API to create `tags.` there you can supply `name` and get an `uid` back then use that  `uid` set in `TagHeaderUid` in subsequent chunks
- just send the first `chunk` without `TagHeaderUid` set, then get back a uid in return header, use that in subsequent chunks (this is where the unnamed_tags_ will be used

The first one is recommended way.
Hope this clarifies .